packages/material-ui/src/Portal/Portal.js

```diff
@@ -19,22 +19,30 @@ const useEnhancedEffect = typeof window !== 'undefined' ? React.useLayoutEffect
 const Portal = React.forwardRef(function Portal(props, ref) {
   const { children, container, disablePortal = false, onRendered } = props;
   const [mountNode, setMountNode] = React.useState(null);
-  const childRef = React.useRef(null);
-  const handleRef = useForkRef(children.ref, childRef);
+  const handleRef = useForkRef(children.ref, ref);
 
   useEnhancedEffect(() => {
     if (!disablePortal) {
       setMountNode(getContainer(container) || document.body);
     }
   }, [container, disablePortal]);
 
-  React.useImperativeHandle(ref, () => mountNode || childRef.current, [mountNode]);
+  useEnhancedEffect(() => {
```

Comment:
Why can't we use `useImperativeHandle` here?

Reply:
Yes, probably with the new if(!node) logic. Will try +1

Reply:
Oh no, we can't. my bad. We still need to ignore the case where disablePortal is false.

Reply:
Could you elaborate on that? Right now we handle a ref in the commit phase. If it's not possible to handle in the ref phase we need to name it differently. Otherwise it might create confusing bugs because it's expected to be updated during ref phase when it only happens during commit.

Reply:
Yes, with pleasure. The problem is in the case where disablePortal is true. The classic ref logic will try to set the value to the DOM node during the first ref phase. However, the useImperativeHandle logic also tries to set the value to null. It creates a conflict. I believe it can't work.

Also, I'm wondering does it care in which phase the ref value is set in our case? Previously, the ref was set to null during the ref phase (to accommodate SSR). Now, the ref isn't called. In both cases, people can't rely on the ref value to be defined, they have to handle this undefined case. They have to wait for the actual value, after the first commit, ref, layout and effect phases.